menas/src/main/scala/za/co/absa/enceladus/menas/services/DatasetService.scala

```diff
@@ -137,111 +162,128 @@ class DatasetService @Autowired() (datasetMongoRepository: DatasetMongoRepositor
   }
 
   private def validateConformanceRules(conformanceRules: List[ConformanceRule],
-                                       maybeSchema: Future[Option[Schema]],
-                                       validations: Future[Validation]): Future[Validation] = {
-    val maybeFields = maybeSchema.map { case Some(x) => x.fields.flatMap(f => f.getAllChildren :+ f.getAbsolutePath).toSet }
-    val accumulator = Tuple2(validations, maybeFields)
+                                       maybeSchema: Future[Option[Schema]]): Future[Validation] = {
+
+    val maybeFields = maybeSchema.map {
+      case Some(x) => x.fields.flatMap(f => f.getAllChildren :+ f.getAbsolutePath).toSet
+      case None => Set.empty[String]
+    }
+    val accumulator = RuleValidationsAndFields(Seq.empty[Future[Validation]], maybeFields)
 
-    conformanceRules.foldLeft(accumulator) { case ((validation, currentColumns), conformanceRule) =>
+    val ruleValidationsAndFields = conformanceRules.foldLeft(accumulator) { case (validationsAndFields, conformanceRule) =>
```

Comment:
**Just a note:** This part of code is out of scope of this class, IMO. But it's good enough now.
Time to refactor is, when the validation is applied to standard dataset edits too.

Reply:
Also, Validations need to be moved to Utils. Then this might be moved to DataModel.